### PR TITLE
Add Interspersed proposal

### PR DESCRIPTION
**Pull request by FranzBusch**
# Motivation
Add a new evolution proposal for the interspersed algorithm.

### CODE REVIEW

**Comment by FranzBusch** [comment]
This depends on #258 to be merged

File: Sources/AsyncAlgorithms/Interspersed/AsyncInterspersedSequence.swift [hunk 1]
```diff
@@ -44,7 +55,7 @@ extension AsyncInterspersedSequence: AsyncSequence {
   public typealias Element = Base.Element
 
   /// The iterator for an `AsyncInterspersedSequence` asynchronous sequence.
-  public struct Iterator: AsyncIteratorProtocol {
+  public struct AsyncIterator: AsyncIteratorProtocol {
```
Comment (phausler) [inline Sources/AsyncAlgorithms/Interspersed/AsyncInterspersedSequence.swift:58]: For consistency; other `AsyncSequence` types name this `Iterator`.

**Review by phausler — APPROVED** [review]
overall looks good to me modulo the slight consistency naming nit.

**Comment by FranzBusch** [comment]
@swift-ci test

**Comment by FranzBusch** [comment]
Merging over known CI failure

**Pull request merged**